src/canvas/Document.tsx

```diff
@@ -29,18 +29,59 @@ export class Document implements IDocument {
             throw new Error("Tried to remove path from document, but there are no paths to remove");
         }
         return Boolean(this._patternPaths.pop());
-    }
+    };
+
+    arePatternPiecesEnclosed = (): boolean => {
+        const endpoints = new Array(this._patternPaths.length * 2);
+
+        this._patternPaths.forEach(path => {
+            const points = path.getPoints();
+            endpoints.push(points[0]);
+            endpoints.push(points[points.length - 1]);
+        });
+
+        for(let i = 0; i < endpoints.length; i++) {
+            const point = endpoints[i];
+            if (!point) {
+                continue;
+            }
+
+            let foundMatch = false;
+            for (let j = i + 1; j < endpoints.length && !foundMatch; j++) {
```

Comment:
The algorithm does not accept : 
- cases where an odd number of paths are meeting at one point (see picture: ![image](https://user-images.githubusercontent.com/34172843/89948688-68281700-dbdb-11ea-90c0-079fb6c714d8.png))
- cases where a path's endpoints lie on another path, but not its endpoint (see picture: 
![image](https://user-images.githubusercontent.com/34172843/89948946-dec51480-dbdb-11ea-8d00-47cee9ad8752.png)). 

I think the second one is ok, but the first one I feel like is something happens in valid patterns.

Reply:
Thanks for finding this. I fixed the first case. I am not sure about whether we should consider the second case valid, which is why I didn't implement it.

Reply:
The last case would be better covered by having symmetry implemented.

Reply:
Yeah I wasn't sure either if it should be valid. It might be necessary to instruct the user to avoid these if we won't allow them, I guess users might do something like this and the app will refuse it they might confused. Maybe not for the MVP though.

Reply:
I suggest we document this in the code, and add a task to the post-mvp backlog to either handle this case differently or add instructions for the users.

Reply:
Yeah, I was thinking some more about patterns where this case might occur and I did think of some. So, I am working on implementing it.

Reply:
I added a Todo here. I do think it is something we should implement, but I will do it in another pr.